core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -64,9 +64,12 @@ protected BaseMetadataTable(TableOperations ops, Table table, String name) {
    */
   static PartitionSpec transformSpec(Schema metadataTableSchema, PartitionSpec spec) {
     PartitionSpec.Builder identitySpecBuilder =
-        PartitionSpec.builderFor(metadataTableSchema).checkConflicts(false);
+        PartitionSpec.builderFor(metadataTableSchema)
+            .withSpecId(spec.specId())
+            .checkConflicts(false);
     for (PartitionField field : spec.fields()) {
-      identitySpecBuilder.add(field.fieldId(), field.name(), Transforms.identity());
+      identitySpecBuilder.add(
+          field.fieldId(), field.fieldId(), field.name(), Transforms.identity());
```

Comment:
Shouldn't this be field.sourceId(), field.fieldId()

Reply:
This makes sense to me.

Reply:
Let me resolve this one.